test/parallel/test-assert-deep.js

```diff
@@ -466,4 +466,10 @@ assertOnlyDeepEqual(
   assertDeepAndStrictEqual(m3, m4);
 }
 
+// Handle NaN
+{
+  assert.throws(() => assert.deepEqual(NaN, NaN), assert.AssertionError);
+  assert.doesNotThrow(() => assert.deepStrictEqual(NaN, NaN));
```

Comment:
Since tests are a form of documentation, maybe add an additional test case or two that demonstrates the utility a bit more:

```js
assert.doesNotThrow(() => { assert.deepStrictEqual({ a: NaN }, { a: NaN });
assert.doesNotThrow(() => { assert.deepStrictEqual( [ 1, 2, NaN, 4 ], [ 1, 2, NaN, 4 ]);
```